internal/test/types/types.go

```diff
@@ -227,56 +132,56 @@ func (w *Windows) GetSSHClient() error {
 
 	config := &ssh.ClientConfig{
 		User:            w.Credentials.GetUserName(), //TODO: Change this to make sure that this works for Azure.
-		Auth:            []ssh.AuthMethod{ssh.Password(w.Credentials.GetPassword())},
+		Auth:            []ssh.AuthMethod{ssh.PublicKeys(w.Credentials.GetSSHKey())},
 		HostKeyCallback: ssh.InsecureIgnoreHostKey(),
 	}
 
-	var err error
-	var sshClient *ssh.Client
-	// Retry if we are unable to create a client as the VM could still be executing the steps in its user data. We
-	// cannot reuse the entries in the retry package as they are too granular.
-	for retries := 0; retries < 10; retries++ {
-		sshClient, err = ssh.Dial("tcp", w.Credentials.GetIPAddress()+":22", config)
-		if err == nil {
-			break
-		}
-		log.Printf("warning - SSH dial %s error: %v", w.Credentials.GetIPAddress(), err)
-		time.Sleep(1 * time.Minute)
-	}
-
+	sshClient, err := ssh.Dial("tcp", w.Credentials.GetIPAddress()+":22", config)
```

Comment:
w.SSHClient can be assigned here

Reply:
checking for error before setting the value.